### PR TITLE
Add clear controller feature

### LINKED ISSUE #382: Add controller's clear() method
**Is your feature request related to a problem? Please describe.**
I use FleatherEditor to send messages like in a normal chat app, so once the message is sent I would like to have an handy method to clear the whole edited message, instead of using [this workaround](https://github.com/fleather-editor/fleather/issues/265#issuecomment-1925615845)

**Describe the solution you'd like**
Instead I would like to be able to do:

`controller.clear();
`

similar to [TextEditingController.clear()](https://api.flutter.dev/flutter/widgets/TextEditingController/clear.html)

### PR DESCRIPTION
**Pull request by Amir-P**
Fixes #382

### CODE REVIEW

File: packages/fleather/lib/src/widgets/controller.dart [hunk 1]
```diff
@@ -311,6 +313,32 @@ class FleatherController extends ChangeNotifier {
       composing: TextRange.empty,
     );
   }
+
+  /// Clear the controller state.
+  ///
+  /// It creates a new empty [ParchmentDocument] and a clean edit history.
+  /// The old [document] will be closed if [closeDocument] is true.
+  ///
+  /// Calling this will notify all the listeners of this [TextEditingController]
```
Comment (amantoux) [inline packages/fleather/lib/src/widgets/controller.dart:322]: You mean `FleatherController`?

File: packages/fleather/test/widgets/controller_test.dart [hunk 2]
```diff
@@ -274,6 +274,23 @@ void main() {
       // expect(controller.lastChangeSource, ChangeSource.local);
     });
 
+    test('clear', () {
```
Comment (amantoux) [inline packages/fleather/test/widgets/controller_test.dart:277]: Shouldn't we test with `closeDocument=false`?

**Review by amantoux — CHANGES_REQUESTED** [review]
It's look good to me overall, there are some nitpicks
Can you please take a look?

File: packages/fleather/test/widgets/controller_test.dart (hunk 2 above)
Comment (Amir-P) [reply packages/fleather/test/widgets/controller_test.dart]: Yea, I forgot about that!

**Review by amantoux — APPROVED** [review]
LGTM

**Pull request merged**